s2fft/precompute_transforms/construct.py

```diff
@@ -193,39 +369,103 @@ def wigner_kernel(
         forward (bool, optional): Whether to provide forward or inverse shift.
             Defaults to False.
 
+        mode (str, optional): Whether to use FFT approach or manually compute each element.
+            {"auto", "manual", "irfft"}. Defaults to "auto" which will detect the
+            most appropriate recursion given the parameter configuration.
+
         using_torch (bool, optional): Desired frontend functionality. Defaults to False.
 
     Returns:
         np.ndarray: Transform kernel for Wigner transform.
     """
+    if mode.lower() == "fft" and sampling.lower() not in ["mw", "mwss", "dh"]:
```

Comment:
Again, I don't think we should switch the mode if auto is not selected.  I would just fail with an error here.  Then the user can correct their settings and rerun.

Reply:
I still think that we shouldn't be intentionally allowing bad configuration, but I've changed this to raise a ValueError instead.